cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os/user"
+	"regexp"
+	"strings"
+	"syscall"
+)
+
+// FindRecursive iterate in a directory (path) showing permissions in a recursive way
+/* path  - where to look for files and directories
+ * level - used to print the output in a hierarchical way
+ */
+func FindRecursive(path, level string) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// In verbose mode we are printing the output in a hierarchical way
+	if Verbose {
+		level = level + "  "
+	}
+
+	for _, f := range files {
+		name := f.Name()
+		relativePath := strings.TrimPrefix(strings.Join([]string{path, name}, "/"), Directory)
+		fullPath := strings.Join([]string{path, name}, "/")
+
+		match, _ := regexp.MatchString(Exclude, relativePath)
+		if match {
+			if Verbose {
+				fmt.Printf("\x1b[33m%sExcluding %s\n\x1b[0m", level, name)
+			}
+		} else {
+			mode := f.Mode()
+			currentOwner, _ := user.LookupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Uid))
+			currentGroup, _ := user.LookupGroupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Gid))
+			currentPerm := mode.Perm().String()
+			hasCorrectOwner := checkOwner(currentOwner.Username, DefaultOwner)
+			hasCorrectGroup := checkOwner(currentGroup.Name, DefaultGroup)
+
+			if mode.IsRegular() && (f.Name()[0:1] != "." || ShowHidden) {
```

Comment:
A few things:
 - You already assigned `f.Name()` to `name`
 - I would early discard hidden files, .e.g. after checking it doesn't match de excluded files, do:
```
if !ShowHidden && strings.HasPrefix(name, ".") {
    continue
}
```
That way you simply ignore them without getting all the properties without showing them
 - Given my previous comment, maybe that flag could be renamed to `ignoreHidden` ? (not a strong opinion though).